### DISCUSSION TITLE
NUCLEO-F429ZI SPI1 & SPI3 parallel

### DISCUSSION THREAD

**Discussion by Stebuson** [stm32duino/Arduino_Core_STM32#1963] [Q&A] [answered]
Hi
I have a NUCLEO-F429ZI board, use PlatformIO and need two SPI Busses. So I initialised and used SPI1 and SPI3 as follows:
`const SPISettings spi_mot_settings(5000000, MSBFIRST, SPI_MODE3);`
`const SPISettings spi_enc_settings(2600000, MSBFIRST, SPI_MODE3);`
`SPIClass SPI_1(D11, D12, D13); // SPI1 mosi, miso, sclk`
`SPIClass SPI_3(D22, D25, D23); // SPI3 mosi, miso, sclk`
`SPI_1.begin();`
`SPI_3.begin();`
`SPI_1.beginTransaction(spi_mot_settings);`
`SPI_3.beginTransaction(spi_enc_settings);`
But it worked not right. Both transfers were visible on both busses.

I got it running with the following:
`SPIClass SPI_3(pinNametoDigitalPin(PB_5_ALT1), pinNametoDigitalPin(PB_4_ALT1), pinNametoDigitalPin(PB_3_ALT1)); // SPI3 mosi, miso, sclk; (D22, D25, D23)`

Does it have to be so complicated or is there an easier way?

**Comment by fpistm** [comment, marked as answer]
`SPIClass SPI_3(PB5_ALT1, PB4_ALT1, PB3_ALT1); // SPI3 mosi, miso, sclk; (D22, D25, D23)`

**Reply by Stebuson** [reply]
Thank you. It works.

I had only tested
`SPIClass SPI_3(PB_5_ALT1, PB_4_ALT1, PB_3_ALT1);`
but it won't work

**Reply by fpistm** [reply]
`PY_n_ALT1` is PinName while `PYn_ALT1` is pin number. Pin number is used for all Standard Arduino API (equivalent to Dx, x).
